project/api/models.py

```diff
@@ -0,0 +1,95 @@
+"""
+Chron.
+Copyright (C) 2018 Alisa Belyaeva, Ata Ali Kilicli, Amaury Martiny,
+Daniil Mordasov, Liam O’Flynn, Mikhail Orlov.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+"""
+
+from django.db import models
+from polymorphic.models import PolymorphicModel
+from colorfield.fields import ColorField
+
+from .ltree import LtreeField
+
+# Create your models here.
+class PoliticalEntity(models.Model):
+    """
+    A 1-1 mapping between a https://www.wikidata.org/wiki/Q56061, and a PK in our db.
+    Holds an additional color information.
+    """
+
+    wikidata_id = models.PositiveIntegerField(primary_key=True)  # Excluding the Q
```

Comment:
I am not sure we would be able to achieve uniqueness for `wikidata_id`  field.
It's better to have our own id. (I've changed my original pov)

For regular countries we store _`administrative territorial entity`_ in that field.
I hope, that there is no entities which has multiple dates of inception and dissolution. But we should keep that in mind.
For disputed territories we store _`geographic region`_ or similar. 
* Hong Kong has a great [history](https://en.wikipedia.org/wiki/History_of_Hong_Kong). This lands were invaded by UK in 1842 later same year became a colony. HK was occupied by Japan during WWII
* Manchuria was invaded more than once by Japan and USSR

Reply:
I'm not sure I understand. I think we can keep uniqueness of wikidata_id.

- HK today: https://www.wikidata.org/wiki/Q8646, but it follows British HK https://www.wikidata.org/wiki/Q1054923. And previous political entities in HK will have different wikidata_ids. Concerning WW2, British HK (Q1054923) will have an IPR (colony) with `start_` and `end_dates`, with a hole in the middle 1941-1945. And it will have a DPR with Japan during this hole.
- Manchuria: idem. We follow wikidata's id. But we can add DPR with `start_date` and `end_date` to show occupation by different entities.

Reply:
What is the difference between disputed status and DPR? How we decide what is DPR and what is disputed?
What is stored in wikidata_id for disputed territories?

In my vision, HK occupated by Japan is no longer British HK and it is a separate PE with it's own government. In this case I think it should be marked as disputed between Japan and British HK.
It should be colored in a different way also (we don't store colors for DPR, only for PE).

>Manchuria: idem.
It's not a country, it's a land. It does not have any wikidata ids except geographic region Q81126
In both cases, when Manchuria was invaded we should mark it as a disputed territory, right?

Reply:
- HK invaded by Japan in 1941, it's clearly a DPR. A disputed territory is when 2 entities claim ownership of said territory. In the case of HK, it's stated in the wikipedia article that British "surrendered", I don't think there is any dispute here.
  - Since it's a DPR, after 1941, HK will have the color of Japan on the zoomed-out view.
  - I agree that wikidata lacks this info, I asked [here](https://opendata.stackexchange.com/questions/13554/wikidata-how-to-represent-british-hong-kongs-under-japanese-occupation). One solution would be what you said: to have a new Wikidata entry called "HK under Japan", from 1941-1945. But I wish to do that on the Wikidata level, not in our db.
- I see the problem. Manchuria was Russian in 1904, and was Japanese in 1905. However in between (i.e. during the war), it was disputed. Is that what you mean? I see 2 solutions:
  - easy one: we say that it was Russian until 1905.
  - hard one: I still maintain that Manchuria should have it's own unique wikidata_id. But I also agree that a `disputed` field on PE will not work. We could say that Manchuria has 2 DPR between 1904-1905, and this means that it's disputed. What do you think?

Reply:
> Maybe we should remove disputed field from PE and allow to have several DPR or IPR that will describe the situation.

Yes, I agree with that.